numpy/lib/tests/test_function_base.py

```diff
@@ -1233,6 +1233,70 @@ def test_empty(self):
         assert_array_equal(a, np.array([0]))
         assert_array_equal(b, np.array([0, 1]))
 
+    def test_optim_numbins(self):
+        """
+        Check that the number of bins returned matches up to expectations.
+        FD and scott depend on the data
+        sturges and rice do not
+        Auto = max(Sturges, FD)
+        """
+        estimator_list = ['fd', 'scott', 'rice', 'sturges']
+
+        # check it can deal with empty data
+        for estimator in estimator_list:
+            # change to self.test_empty after merge accepted
+            # otherwise too much effort to keep updating PR
+            a, b = histogram([], estimator)
+            assert_array_equal(a, np.array([0]))
+            assert_array_equal(b, np.array([0, 1]))
+
+
+        # check that unknown strings throw a Value Error too
+        check_list = ['mad', 'freeman', 'histograms', 'IQR']
+        for estimator in check_list:
+            try:
+                a, b = histogram([1,2,3], estimator)
+            except ValueError:
+                pass
+            else:
+                raise AssertionError("np.histogram ran when passed "
+                                     "unknown automatic method: {}".format(estimator))
+
+        # some basic sanity checking, with some fixed data. Checking for the correct number of bins
+        basic_test = {50:   {'fd': 4,  'scott': 4,  'rice': 8,  'sturges': 7,  'auto': 7},
+                      500:  {'fd': 8,  'scott': 8,  'rice': 16, 'sturges': 10, 'auto': 10},
+                      5000: {'fd': 17, 'scott': 17, 'rice': 35, 'sturges': 14, 'auto': 17}}
+
+        for testlen, expectedResults in basic_test.items():
+            # create some sort of non uniform data to test with (2 peak uniform mixture)
+            x1 = np.linspace(-10, -1, testlen/5 * 2)
+            x2 = np.linspace(1,10, testlen/5 * 3)
+            x = np.hstack((x1, x2))
+            for estimator, numbins in expectedResults.items():
+                a, b = np.histogram(x, estimator)
+                assert(len(a) == numbins)
```

Comment:
Can you replace all `assert(...)` with the function `assert_equal(len(a), numbins)`, etc. `assert`s may be removed when running optimized and then the tests would be useless.
Could make it a class and split up the tests more, but don't worry about it.